src/gamma_inc.jl

```diff
@@ -143,6 +155,230 @@ function rgammax(a::Float64,x::Float64)
+
+Compute ``log(\\Gamma(1+a))`` for -1 < a <= 1.
+"""
+function loggamma1p(a::Float64)
+  
+    if a >= 0.6
+        x = a - 1.0
+        w = @horner(x, .422784335098467E+00, .848044614534529E+00, .565221050691933E+00, .156513060486551E+00, .170502484022650E-01, .497958207639485E-03) / @horner(x, 1.0, .124313399877507E+01, .548042109832463E+00, .101552187439830E+00, .713309612391000E-02, .116165475989616E-03)
+        return x*w
+    else
+        w = @horner(a, .577215664901533E+00, .844203922187225E+00, -.168860593646662E+00, -.780427615533591E+00, -.402055799310489E+00, -.673562214325671E-01, -.271935708322958E-02) / @horner(a, 1.0, .288743195473681E+01, .312755088914843E+01, .156875193295039E+01, .361951990101499E+00, .325038868253937E-01, .667465618796164E-03)
+        return -a*w
+    end
+end
+
+"""
+    chepolsum(n,x,a)
+
+Computes a series of Chebyshev Polynomials given by : a[0]/2 + a[1]T1(x) + .... + a[n]Tn(X)
+"""
+function chepolsum(x::Float64, a::Array{Float64,1})
+    n = length(a) - 1
+  
+    if n == 0
+        return a[1]/2.0
+    elseif n == 1
+        return a[1]/2.0 + a[1]*x
+    else
+        tx = 2*x
+        r = a[n]
+        h = a[n-1] + r*tx
+        for k = n-2:-1:1
+            s=r
+            r=h
+            h=a[k]+r*tx-s
+        end
+        return a[0]/2.0 - r + h*x
+    end
+end
+"""
+    stirling(x)
+
+Compute the Stirling Series for given `x`.
+"""
+function stirling(x::Float64)
+  
+    a = zeros(18) 
+    if x < floatmin(Float64)*1000.0
+        return floatmax(Float64)/1000.0
+    elseif x < 1
+        return loggamma1p(x) - (x+0.5)*log(x)+x - log(sqrt(2*pi))
+    elseif x < 2
+        return loggamma1p(x-1) - (x-0.5)*log(x)+x - log(sqrt(2*pi))
+    elseif x < 3
+        return loggamma1p(x-2) - (x-0.5*log(x)+x  - log(sqrt(2*pi))+log(x-1))
+    elseif x < 12
+        a[1]=1.996379051590076518221
+        a[2]=-0.17971032528832887213e-2
+        a[3]=0.131292857963846713e-4
+        a[4]=-0.2340875228178749e-6
+        a[5]=0.72291210671127e-8
+        a[6]=-0.3280997607821e-9
+        a[7]=0.198750709010e-10
+        a[8]=-0.15092141830e-11
+        a[9]=0.1375340084e-12
+        a[10]=-0.145728923e-13
+        a[11]=0.17532367e-14
+        a[12]=-0.2351465e-15
+        a[13]=0.346551e-16
+        a[14]=-0.55471e-17
+        a[15]=0.9548e-18
+        a[16]=-0.1748e-18
+        a[17]=0.332e-19
+        a[18]=-0.58e-20
+        z=18.0/(x*x)-1.0
+        return chepolsum(z,a)/(12.0*x)
+    else
+        c = zeros(7)
+        z = 1.0/(x*x)
+        if x < 1000
+            return @horner(z, 0.25721014990011306473e-1, 0.82475966166999631057e-1, -0.25328157302663562668e-2, 0.60992926669463371e-3, -0.33543297638406e-3, 0.250505279903e-3)/((0.30865217988013567769 + z)/x)
+        else
+            return (((-z/1680.0+1.0/1260.0)*z-1.0/360.0)*z+1.0/12.0)/x
+        end
+    end
+end
+"""
+    gammax(x)
+
+gammax(x) = ``e^{stirling(x)}`` if x>0
+else ``\\Gamma(x)/(e^{-x + (x-0.5)*log(x)}/\\sqrt{2 \\pi}``
```

Comment:
These are the same: `e^{stirling(x)}` is an approximation to ``\\Gamma(x)/(e^{-x + (x-0.5)*log(x)}/\\sqrt{2 \\pi}``

Reply:
Yes, but exp(stirling(x)) gives different value than \\Gamma(x)/(e^{-x + (x-0.5)*log(x)}/\\sqrt{2 \\pi} when x>3.
It might be some domain related precision issue.

Reply:
Now that we've figured out the issue with `stirling`, this can be updated.